api_app/connectors_manager/connectors/misp.py

```diff
@@ -1,10 +1,94 @@
 # This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
 # See the file 'LICENSE' for copying permission.
 
+from api_app import helpers
+import pymisp
 
-from ..classes import Connector
+from api_app.connectors_manager.classes import Connector
+
+
+INTELOWL_MISP_TYPE_MAP = {
+    "ip": "ip-src",
+    "domain": "domain",
+    "url": "url",
+    # "hash" (checked from helpers.get_hash_type)
+    "general": "text",  # misc field, so keeping text
+    "file": "filename",
+}
 
 
 class MISP(Connector):
+    def set_params(self, params):
+        self.ssl_check = params.get("ssl_check", True)
+        self.debug = params.get("debug", False)
+        self.__url_name = self._secrets["url_key_name"]
+        self.__api_key = self._secrets["api_key_name"]
+
+    @property
+    def _event_obj(self):
+        obj = pymisp.MISPEvent()
+        obj.info = f"Intelowl Analysis: {self._job.observable_name}"
+        obj.distribution = 0  # your_organisation_only
+        obj.threat_level_id = 4  # undefined
+        obj.analysis = 2  # completed
+        obj.add_tag("intelowl")
+
+        # Add tags from Job
+        for tag in self._job.tags.all():
+            obj.add_tag(f"{tag.label}:{tag.color}")
+
+        return obj
+
+    @property
+    def _base_attr_obj(self):
+        if self._job.is_sample:
+            type = "file"
+            value = f"{self._job.file_name} ({self._job.file_mimetype})"
```

Comment:
for the files, maybe it is better to use the attribute `filename|md5`. Otherwise we could add an additional attribute for the `md5` too.

Also, for the mimetype, we could add a separated attribute `mime-type`

See: https://www.circl.lu/doc/misp/categories-and-types/

Reply:
Done